### PR TITLE
Add full_table_name and avro_schema_prefix options

### LINKED ISSUE #9450: Topic Name configurability
In 21.1 we added several options that give the user additional power over kafka topic naming: the ability to specify a topic prefix (already documented), the ability to use fully qualified table name.

**Fully Qualified Name**
Add option to https://www.cockroachlabs.com/docs/v20.2/create-changefeed.html#options 
- option: full_table_name
- value: N/A
- Description: Use fully qualified table name for topic creation rather than the default (table name).  EXAMPLE: "movr.public.drivers" instead of "drivers".

**Topic Creation**
The nuances of topic creation haven't been well documented, and with the new options, its probably worth it to have a section on topics. I'm not sure where the best place for this is, probably in https://www.cockroachlabs.com/docs/v20.2/create-changefeed.html#options in a new section called "Topic Naming", with the following information:

By default, a kafka topic has the same name as the table that a changefeed was created on. If a changefeed was created on multiple tables, the changefeed will write to multiple topics corresponding to those table names. You can specify a topic prefix or use the full_table_name option to modify this. 

You can either manually create the topic in your kafka cluster before starting the changefeed, or the topic will be automatically created when the changefeed connects to your kafka cluster. Note: you must have the kafka cluster setting [auto.create.topics.enable](https://kafka.apache.org/documentation/#brokerconfigs_auto.create.topics.enable) set to True for automatic topic creation. 

Kafka has the following topic [limitations](https://github.com/apache/kafka/blob/0.10.2/core/src/main/scala/kafka/common/Topic.scala#L24) :
- Legal characters: numbers, letters, and [._-]
- Maximum character length of name: 249
- Topics with a period (.) and underscore (_) can collide on internal kafka data structures, so they advise using either, but not both 
For characters not accepted by Kafka, we encode them as unicode characters.

### LINKED ISSUE #10054: changefeedccl: Add WITH avro_schema_prefix option to changefeeds in avro format
PR: https://github.com/cockroachdb/cockroach/pull/59710

From release notes:
> Added `WITH avro_schema_prefix` option to Avro [changefeeds](../v21.1/create-changefeed.html) to prevent collisions in a shared schema registry [#59710][#59710] {% comment %}doc{% endcomment %}

### PR DESCRIPTION
**Pull request by gemma-shay**
-Add the `WITH full_table_name` option to the Options table
-Add the `WITH avro_schema_prefix=[prefix]` option to the Options table
-Add Topic Naming explanation/limitations section

Closes #9450
Closes #10054

### CODE REVIEW

File: v21.1/create-changefeed.md [hunk 1]
```diff
@@ -148,6 +150,23 @@ CockroachDB Type | Avro Type | Avro Logical Type
...
+{{site.data.alerts.callout_info}}
+You must have the Kafka cluster setting `auto.create.topics.enable` set to `True` for automatic topic creation.
+{{site.data.alerts.end}}
+
+Kafka has the following topic limitations:
+
+- [Legal characters](https://github.com/apache/kafka/blob/0.10.2/core/src/main/scala/kafka/common/Topic.scala#L29) are numbers, letters, and `[._-]`.
+- The maximum character length of a topic name is 249.
+- Topics with a period (`.`) and underscore (`_`) can collide on internal Kafka data structures, so you should use either but not both.
+- Characters not accepted by Kafka can be encoded as unicode characters.
```
Comment (HonoreDB) [inline v21.1/create-changefeed.md:168]: Suggest clarifying that Cockroach will do this automatically.

File: v21.1/create-changefeed.md [hunk 2]
```diff
@@ -148,6 +150,23 @@ CockroachDB Type | Avro Type | Avro Logical Type
...
 [`JSONB`](jsonb.html) | [`STRING`](https://avro.apache.org/docs/1.8.1/spec.html#schema_primitive) |
 
+#### Topic Naming
+
+By default, a Kafka topic has the same name as the table that a changefeed was created on. If a changefeed was created on multiple tables, the changefeed will write to multiple topics corresponding to those table names. You can specify a [topic prefix](#query-parameters) or use the [`full_table_name` option](#options) to modify this.
+
+You can either manually create a topic in your Kafka cluster before starting the changefeed, or the topic will be automatically created when the changefeed connects to your Kafka cluster. 
+
+{{site.data.alerts.callout_info}}
+You must have the Kafka cluster setting `auto.create.topics.enable` set to `True` for automatic topic creation.
```
Comment (HonoreDB) [inline v21.1/create-changefeed.md:160]: When using automatic topic creation, the topic will not be created until the changefeed sends its first message. If you create the consumer before that, you will also need the `allow.auto.create.topics` consumer setting.

**Review by HonoreDB — APPROVED** [review]
Looks great!

File: v21.1/create-changefeed.md (hunk 2 above)
Comment (gemma-shay) [reply v21.1/create-changefeed.md:160]: Done.

**Review by lnhsingh — COMMENTED** [review]
Small nit, otherwise LGTM! Nice job

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/10292#-:-MYLYOzo5W-23BR6xFWh:b89wknc)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @amruss, @gemma-shay, and @HonoreDB)

___
*[v21.1/create-changefeed.md, line 168 at r1](https://reviewable.io/reviews/cockroachdb/docs/10292#-MXmn3Bl7Z-MyVjMoCtP-r1-168:-MYLY-dR1hAti4fh-A6o:b-xir4yg) ([raw file](https://github.com/cockroachdb/docs/blob/97dfc0350f7a9d06ed621cbe42148c2e048b2152/v21.1/create-changefeed.md#L168)):*
<details><summary><i>Previously, HonoreDB (Aaron Zinger) wrote…</i></summary><blockquote>

Suggest clarifying that Cockroach will do this automatically.
</blockquote></details>

^ Add Aaron's suggestion + "Cockroach" should be "CockroachDB"
___
*[v21.1/create-changefeed.md, line 113 at r2](https://reviewable.io/reviews/cockroachdb/docs/10292#-MYLWAvr9DrfQnEl5rcW:-MYLWAvr9DrfQnEl5rcX:bcbj1li) ([raw file](https://github.com/cockroachdb/docs/blob/7fb023731f4314365b9bdb3618aeb7edb9cc41e5/v21.1/create-changefeed.md#L113)):*
> ```Markdown
> `initial_scan` / `no_initial_scan` | N/A |  Control whether or not an initial scan will occur at the start time of a changefeed. `initial_scan` and `no_initial_scan` cannot be used simultaneously. If neither `initial_scan` nor `no_initial_scan` is specified, an initial scan will occur if there is no `cursor`, and will not occur if there is one. This preserves the behavior from previous releases.<br><br>Default: `initial_scan` <br>If used in conjunction with `cursor`, an initial scan will be performed at the cursor timestamp. If no `cursor` is specified, the initial scan is performed at `now()`.
> `full_table_name` | N/A |<span class="version-tag"> New in v21.1: </span> Use fully-qualified table name in topics, subjects, schemas, and record output instead of the default table name. This can prevent unintended behavior when the same table name is present in multiple databases. <br><br>Example: `CREATE CHANGEFEED FOR foo... WITH full_table_name` will create the topic name `defaultdb.public.foo` instead of `foo`.
> `avro_schema_prefix` |Schema prefix name               | <span class="version-tag"> New in v21.1: </span> Use fully-qualified schema name for a table instead of the default table name. This allows multiple databases or clusters to share the same schema registry when the same table name is present in multiple databases.<br><br>Example: `CREATE CHANGEFEED FOR foo WITH format=experimental_avro, confluent_schema_registry='registry_url', avro_schema_prefix='super'` will register subjects as `superfoo-key` and `superfoo-value` with the namespace `super`.
> ```

nit: add a space before `Schema prefix name` (even though this looks like it renders ok, it's good to keep the markdown consistent)

**Review by amruss — APPROVED** [review]
LGTM

**Review by lnhsingh — COMMENTED** [review]
:lgtm_strong: 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/10292#-:-MYeQcC24RxKJbcYWisU:b-uwkowd)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @gemma-shay and @HonoreDB)

**Pull request merged**